src/libs/Navigation/AppNavigator/MainDrawerNavigator.js

```diff
@@ -1,49 +1,87 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
 import {createDrawerNavigator} from '@react-navigation/drawer';
+import {withOnyx} from 'react-native-onyx';
 
 import withWindowDimensions, {windowDimensionsPropTypes} from '../../../components/withWindowDimensions';
+import FullScreenLoadingIndicator from '../../../components/FullscreenLoadingIndicator';
+import {getLastAccessedReport} from '../../reportUtils';
 import styles, {
     getNavigationDrawerType,
     getNavigationDrawerStyle,
 } from '../../../styles/styles';
+import ONYXKEYS from '../../../ONYXKEYS';
+import compose from '../../compose';
+import SCREENS from '../../../SCREENS';
 
 // Screens
 import SidebarScreen from '../../../pages/home/sidebar/SidebarScreen';
 import ReportScreen from '../../../pages/home/ReportScreen';
 
 const propTypes = {
+    // Available reports that would be displayed in this navigator
+    reports: PropTypes.objectOf(PropTypes.shape({
+        reportID: PropTypes.number,
+    })),
+
     ...windowDimensionsPropTypes,
 };
 
+const defaultProps = {
+    reports: {},
+};
+
 const Drawer = createDrawerNavigator();
 
-const MainDrawerNavigator = props => (
-    <Drawer.Navigator
-        openByDefault
-        drawerType={getNavigationDrawerType(props.isSmallScreenWidth)}
-        drawerStyle={getNavigationDrawerStyle(
-            props.windowWidth,
-            props.isSmallScreenWidth,
-        )}
-        sceneContainerStyle={styles.navigationSceneContainer}
-        edgeWidth={500}
-        drawerContent={() => <SidebarScreen />}
-    >
-        <Drawer.Screen
-            name="Report"
-            component={ReportScreen}
-
-            // Providing an empty string here will ensure that the ReportScreen does not show as '/r/undefined'
-            // eslint-disable-next-line react/jsx-props-no-multi-spaces
-            initialParams={{reportID: ''}}
-            options={{
+// Decorated to always returning the result of the first call - keeps Screen initialParams from changing
+const getInitialReport = _.once(getLastAccessedReport);
+
+const MainDrawerNavigator = (props) => {
+    // When there are no reports there's no point to render the empty navigator
+    if (_.size(props.reports) === 0) {
+        return <FullScreenLoadingIndicator visible />;
+    }
+
+    const initialReportID = getInitialReport(props.reports).reportID;
```

Comment:
Once reports are loaded initial value is derived and no longer updates.
It doesn't need to update because the Screen would use `initialParams` when it mounts and no longer care if they change

The value returned by `getLastAccessedReport` would be the same as `ONYXKEYS.CURRENTLY_VIEWED_REPORTID`, but if there's no data in Onyx (like after signout -> login) the last seen report by the user will be resolved, this seems a bit better than just using the first report in the list.

This also simplifies other places:
- nothing no longer needs to make sure a `lastViewedReportID` is set so that Navigation can mount
- A chunk of code in `Report.fetchAllReports` was removed as it's no longer necessary

Reply:
Yep that all sounds great to me!